docs/spec.md

```diff
@@ -0,0 +1,406 @@
+# Storacha MCP Server Spec
+
+## Overview
+
+This document outlines the specification for implementing a Model Context Protocol (MCP) server for Storacha storage. This MCP server will provide a standardized interface for AI models and applications to interact with Storacha's decentralized storage capabilities using a simple HTTP-based approach.
+
+## Key Concepts
+
+An MCP server provides a standardized way for AI applications to discover and use external tools and resources. For Storacha, this means:
+
+- AI models can securely access Storacha's decentralized storage
+- Applications can store and retrieve data without custom integration code
+- The MCP standard ensures compatibility with a growing ecosystem of clients
+
+## Storacha MCP Server Architecture
+
+### Basic Flow
+
+```mermaid
+flowchart LR
+    A[AI Application] -->|HTTP Request| B[Storacha MCP Server]
+    B -->|Upload| C[w3up-client]
+    B -->|Retrieve| E[HTTP Gateway]
+    C -->|Store| D[(Decentralized Storage)]
+    D -->|Content-addressed Data| E
+    C -->|Response| B
+    E -->|Response| B
+    B -->|HTTP Response| A
+    
+    classDef client fill:#d4f1f9,stroke:#0078d7,stroke-width:2px
+    classDef server fill:#ffe6cc,stroke:#d79b00,stroke-width:2px
+    classDef storage fill:#d5e8d4,stroke:#82b366,stroke-width:2px
+    classDef gateway fill:#fff2cc,stroke:#d6b656,stroke-width:2px
+    
+    class A client
+    class B server
+    class C,D storage
+    class E gateway
+```
+
+### Core Functionality
+
+The Storacha MCP server will support two primary operations in this first iteration:
+
+1. **Upload**: Store files on Storacha's decentralized storage using the Storacha client
+2. **Retrieve**: Get files from Storacha by their Content ID (CID) through the Storacha HTTP trustless Gateway
+3. **Authentication & Authorization**: Optional authentication via bearer tokens
+
+#### Upload Operation
+
+- **Multiple File Support**: The upload operation can handle one or more files in a single request
+- **Universal File Types**: Supports any file type, which is automatically converted to a blob by the Storacha client
```

Comment:
> automatically converted to a blob

What does this mean?

Reply:
My idea was to let the user upload any format of the file, and then in the mcp server create the Blob object. I guess I should call that Storacha Client Wrapper instead.